### PR TITLE
Fix response structure

### PR DESCRIPTION
**Pull request by ghost**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by bizarreWizard** [comment]
Hi,
The SDK generated based on version 2019-08-01 is broken as we changed the our response structure. In order to fix this version, the schema must be updated accordingly.

**Review by yangyuan — APPROVED** [review]

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 1]
```diff
@@ -894,7 +894,7 @@
           "type": "boolean",
           "description": "Enable hierarchical extraction of Q-A from files and urls. Value to be considered False if this field is not present."
         },
-        "defaultAnswerUsedForExtraction":{
+        "defaultAnswerUsedForExtraction": {
```
Comment (erich-wang) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: pls revert this change

**Review by erich-wang — CHANGES_REQUESTED** [review]

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json [hunk 2]
```diff
@@ -4592,23 +4592,27 @@
...
-          "description": "A system generated property that specifies the addressable path of the users resource."
-        }
-      },
-      "allOf": [
-        {
-          "$ref": "#/definitions/SqlDatabaseResource"
-        },
-        {
-          "$ref": "#/definitions/ExtendedResourceProperties"
+        "resource": {
```
Comment (ryansbenson) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: Is adding 'resource' around all these properties a change to your service or a correction to the swagger documentation?

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json [hunk 3]
```diff
@@ -5904,6 +5952,20 @@
           "items": {
             "$ref": "#/definitions/ExcludedPath"
           }
+        },
+        "compositeIndexes": {
```
Comment (ryansbenson) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: Adding writable properties to an existing response is considered a breaking change and requires a new api version.
https://microsoft.sharepoint.com/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json (hunk 2 above)
Comment (bizarreWizard) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: It's the fix to our service. As the Export template in portal is broken.

Comment (bizarreWizard) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: Adding the "resource" is the the change to this swagger documentation. We shouldn't have response body structure differentiate from the request.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json (hunk 3 above)
Comment (bizarreWizard) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: Given this version is in broken status. And all the generated SDKs cannot work now. We need fix this version rather than bring a new version.

Comment (ryansbenson) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: This is the premise of both my comments really, are you planning to change your service to align with this swagger change or is this swagger change to align with the deployed service?

If your service code/contract isn't changing and this is a fix to your sdk/documentation, then we can allow this through (even though it looks like a breaking change).

Comment (bizarreWizard) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: This swagger is change to align with the deployed service. The service code will not be changed.

**Comment by bizarreWizard** [comment]
@ryansbenson
Can you approve and merge the PR? Thank you very much

**Comment by bizarreWizard** [comment]
Hi, can anyone merge the PR, if no, could you please advice what's blocker?
Thanks!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json [hunk 4]
```diff
@@ -5974,6 +6036,64 @@
...
+          }
+        }
+      }
+    },
+    "SpatialSpec": {
+      "type": "object",
+      "properties": {
+        "path": {
+          "type": "string",
+          "description": "The path for which the indexing behavior applies to. Index paths typically start with root and end with wildcard (/path/*)"
```
Comment (sanjaiganesh) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json]: Add  "pattern":  property to define the regex. helps fail early.. probably at SDK layer.

**Comment by sanjaiganesh** [comment]
Add ARM sign off.. Breaking changes.. but it seems service is behaving this way and just fixing Swagger.

**Review by erich-wang — APPROVED** [review]
